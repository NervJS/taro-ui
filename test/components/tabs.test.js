import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtTabs from '../../.temp/components/tabs/index'

describe('AtTabs Snap', () => {
  const tabList = [
    { title: '标签页1' },
    { title: '标签页2' },
    { title: '标签页3' }
  ]

  it('render initial AtTabs', () => {
    const componet = renderToString(<AtTabs />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props customStyle', () => {
    const componet = renderToString(<AtTabs customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props className', () => {
    const componet = renderToString(<AtTabs className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props tabList', () => {
    const componet = renderToString(<AtTabs tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props current', () => {
    const componet = renderToString(<AtTabs current={2} tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props swipeable', () => {
    const componet = renderToString(<AtTabs swipeable tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props scroll', () => {
    const componet = renderToString(<AtTabs scroll tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props tabDirection', () => {
    const componet = renderToString(
      <AtTabs scroll tabDirection='vertical' tabList={tabList} />
    )
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props animated', () => {
    const componet = renderToString(
      <AtTabs animated={false} tabList={tabList} />
    )
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props height', () => {
    const componet = renderToString(
      <AtTabs height='300px' tabDirection='vertical' tabList={tabList} />
    )
    expect(componet).toMatchSnapshot()
  })
})
