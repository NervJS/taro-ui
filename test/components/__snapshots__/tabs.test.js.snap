// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabs Snap render AtTabs -- props animated 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);transition:unset;height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:300%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props className 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB test\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:0%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props current 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(-200%, 0px, 0px);-webkit-transform:translate3d(-200%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:300%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props customStyle 1`] = `"<div style=\\"height:;color:red;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:0%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props height 1`] = `"<div style=\\"height:300px;\\" class=\\"at-tabs at-tabs--vertical at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(0px, -0%, 0px);-webkit-transform:translate3d(0px, -0%, 0px);height:300px;\\" class=\\"at-tabs__body\\"><div style=\\"height:300%;width:1PX;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props scroll 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--scroll at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" style=\\"height:;\\" scrollX scrollWithAnimation scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-x at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:300%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props swipeable 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:300%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props tabDirection 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--scroll at-tabs--vertical at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" style=\\"height:;\\" scrollY scrollWithAnimation scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(0px, -0%, 0px);-webkit-transform:translate3d(0px, -0%, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:300%;width:1PX;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props tabList 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2<div class=\\"at-tabs__item-underline\\"></div></div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3<div class=\\"at-tabs__item-underline\\"></div></div></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:300%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;

exports[`AtTabs Snap render initial AtTabs 1`] = `"<div style=\\"height:;\\" class=\\"at-tabs at-tabs--horizontal at-tabs--WEB\\"><div id=\\"tabs-AOTU2018\\" class=\\"at-tabs__header\\"></div><div style=\\"transform:translate3d(-0%, 0px, 0px);-webkit-transform:translate3d(-0%, 0px, 0px);height:;\\" class=\\"at-tabs__body\\"><div style=\\"height:1PX;width:0%;\\" class=\\"at-tabs__underline\\"></div></div></div>"`;
